### PR TITLE
Maintain the invariant that started <= max_runs

### PR DESCRIPTION
**Pull request by sfc-gh-anoyes**
Or that max_runs == 0, which apparently means run forever

### CODE REVIEW

**Review by sfc-gh-satherton — APPROVED** [review]

**Comment by sfc-gh-anoyes** [comment]
I ran the tests locally since CI had an issue

File: joshua/joshua_model.py [hunk 1]
```diff
@@ -694,6 +698,7 @@ def should_run_ensemble(tr: fdb.Transaction, ensemble_id: str) -> bool:
                 )
             )
 
+            _decrement(tr, ensemble_id, "started")
```
Comment (sfc-gh-kmakino) [inline joshua/joshua_model.py:701]: If many agents are trying to steal this ensemble at the same time, do they likely conflict to each other and keep retrying? (We have read conflict on `started`)
How can we make sure the first agent gets it?

Comment (sfc-gh-kmakino) [reply joshua/joshua_model.py:701]: Also, `should_run_ensemble` does not start a test. Meaning, when one agent decrements `started`, other agents can still get the same ensemble through `should_run_ensemble` before it gets picked up.

Comment (sfc-gh-anoyes) [reply joshua/joshua_model.py:701]: One of them will get it, and the rest will fail with a conflict (and won't see any work to do when they retry)

**Review by jzhou77 — APPROVED** [review]

**Pull request merged**